File: 0953 - Verifying an Alien Dictionary/solution.py
class Solution:
    def isAlienSorted(self, A: List[str], order: str) -> bool:
        m = {c: i for i, c in enumerate(order)}
        A = [[m[i] for i in s] for s in A]
        return all(s1 <= s2 for s1, s2 in zip(A, A[1:]))

File: 0953 - Verifying an Alien Dictionary/solution.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& A, string order) {
        int m[26];
        for(int i = 0; i < 26; i++) m[order[i]-'a'] = i;
        auto cmp = [&](string& a, string& b){
            for(int i = 0; i < min(a.size(), b.size()); i++)
                if(a[i] != b[i]) return m[a[i]-'a'] < m[b[i]-'a'];
            return a.size() < b.size();
        };
        return is_sorted(A.begin(), A.end(), cmp);
    }
};